Mouse speed UI

laxu opened the pull request:
Implements Mouse speed page UI as per #458

mondalaci commented:
Thank you!

1. I can experience the double scrollbar issue here, too.
2. How about placig the reset button at the bottom of the page? I think it'd be a better place, and consistent with the placement of the Configuration page.
3. The step value of the mouse movement speed sliders should be set to 25 to assure that these values can only be multiplies of 25. Currently, when activating a slider and using the left and right keys, the value is always incremented by 1, and when moving these sliders with the mouse the values are not multiplies of 25.

laxu commented:
Changed the slider step to 25 and moved the reset button to the bottom. I originally put it on top because I thought it might not be found on the bottom but I guess it's ok there.

The double scrollbar is caused by the `.main-page` element being set to `display: inline-block`. I can remove the rule if it doesn't cause any issues on other pages.

mondalaci commented:
I'm surprised about `inline-block` because I did fix it in master yesterday and when merging it to your branch, nothing seemed to happen. Would you please remove it?

mondalaci commented:
Everything seems to be perfect on my side, too now that I cloned the repo again. Not sure what was the problem.

Waiting for the review of @ert78gb before merging this. Thanks for the awesome work!

ert78gb commented on packages/uhk-web/src/app/components/device/mouse-speed/mouse-speed.component.ts:
```diff
@@ -8,5 +17,137 @@ import { Component } from '@angular/core';
+        }));
+    }
+
+    resetToDefault() {
```
pls create a new device Action `ResetMouseSettingsAction' and implement this functionality in the a DeviceEffect. pls use LoadResetUserConfigurationAction from the effect to load the settings in the reducer

laxu replied:
Done.

ert78gb commented on packages/uhk-web/src/app/components/device/mouse-speed/mouse-speed.component.ts:
```diff
@@ -8,5 +17,137 @@ import { Component } from '@angular/core';
+export class MouseSpeedComponent implements OnInit, OnDestroy {
+    public moveProps = [
+        {
+            prop: 'mouseMoveInitialSpeed',
```
The agent currently template driven and I feel if you are create a new component and set property bindings than this part of code should be in the template for the consistency.

laxu replied:
I'd rather keep them in here. I understand it creates more mess in the component itself but the other alternative is just a ton of repeated code in the template. I think this is a more maintainable solution.

ert78gb commented on packages/uhk-web/src/app/components/device/mouse-speed/mouse-speed.component.ts:
```diff
@@ -8,5 +17,137 @@ import { Component } from '@angular/core';
+        this.userConfig$ = this.store.select(getUserConfiguration);
+        this.userConfigSubscription = this.userConfig$.subscribe(config => {
+            this.moveProps.forEach(moveProp => {
+                moveProp.value = config[moveProp.prop] * MOUSE_MOVE_VALUE_MULTIPLIER || 0;
```
@mondalaci I think better place to modify the values in the de/serialization?  If somewhere else we modify this values than we don't have to multiply the values.

mondalaci replied:
It'd be nice to deal with the first-order, non-multiplied values everywhere, and only tell the slider to multiply it if that's what you mean.

laxu replied:
Right now it works so that the slider handles the multiplied value only and the non-multiplied is used everywhere else.

laxu commented on packages/uhk-web/src/app/components/device/mouse-speed/mouse-speed.component.html:
```diff
@@ -2,6 +2,43 @@ <h1>
+        cancelText="No"
+        (confirm)="resetToDefault()">Reset speeds to default
+</button>
+<ng-template #sliderTemplate let-prop="prop" let-sliderSettings="sliderSettings">
```
Done.

mondalaci commented:
Thanks @laxu!

@ert78gb Would you take a final look?

ert78gb commented on packages/uhk-web/src/app/store/effects/device.ts:
```diff
@@ -18,6 +18,7 @@ import {
     ActionTypes,
     ConnectionStateChangedAction,
     HideSaveToKeyboardButton,
+    ResetMouseSpeedSettingsAction,
```
it is an unused import pls delete it or use it in the `ofType`

ert78gb replied:
Everything else is ok. @laxu great job
